geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AttributeIdxStrategy.scala

```diff
@@ -285,8 +292,29 @@ object AttributeIdxStrategy extends StrategyProvider {
         case Cardinality.UNKNOWN => 101 * multiplier
         case Cardinality.LOW     => Long.MaxValue
       }
-    }.sum
-    if (cost == 0) Long.MaxValue else cost // cost == 0 if somehow the filters don't match anything
+    }
+    cost.getOrElse(Long.MaxValue)
+  }
+
+  /**
+    * Checks for attribute filters that we can satisfy using the attribute index strategy
+    *
+    * @param filter filter to evaluate
+    * @return true if we can process it as an attribute query
+    */
+  def attributeCheck(filter: Filter): Boolean = {
+    filter match {
+      case _: And | _: Or => true // note: implies further processing of children
+      case _: PropertyIsEqualTo => true
+      case _: PropertyIsBetween => true
+      case _: PropertyIsGreaterThan | _: PropertyIsLessThan => true
+      case _: PropertyIsGreaterThanOrEqualTo | _: PropertyIsLessThanOrEqualTo => true
+      case _: During |  _: Before | _: After | _: TEquals => true
+      case _: PropertyIsNull => true // we need this to be able to handle 'not null'
+      case f: PropertyIsLike => likeEligible(f)
+      case f: Not =>  f.getFilter.isInstanceOf[PropertyIsNull]
```

Comment:
rewrite your query as less than or equal to? :)
the only NOT query we handle is NOT NULL, as that's a simple scan of the attribute table.
We should probably add some simple filter rewriting up front - things like `dtg < foo AND dtg > bar` don't get handled properly either now. But on the other hand, there are clear work-arounds.